packages/block-editor/src/components/block-list/block.js

```diff
@@ -275,32 +296,43 @@ function BlockListBlock( {
 	 *
 	 * @param {KeyboardEvent} event Keydown event.
 	 */
-	const deleteOrInsertAfterWrapper = ( event ) => {
+	const onKeyDown = ( event ) => {
 		const { keyCode, target } = event;
 
-		// These block shortcuts should only trigger if the wrapper of the block is selected
-		// And when it's not a multi-selection to avoid conflicting with RichText/Inputs and multiselection.
-		if (
-			! isSelected ||
-			target !== wrapper.current ||
-			isLocked
-		) {
-			return;
-		}
+		// ENTER/BACKSPACE Shortcuts are only available if the wrapper is focused
+		// and the block is not locked.
+		const canUseShortcuts = (
+			isSelected &&
+				! isLocked &&
+				( target === wrapper.current || target === breadcrumb.current )
+		);
+		const isEditMode = ! isNavigationMode;
 
 		switch ( keyCode ) {
 			case ENTER:
-				// Insert default block after current block if enter and event
-				// not already handled by descendant.
-				onInsertDefaultBlockAfter();
-				event.preventDefault();
+				if ( canUseShortcuts && isEditMode ) {
```

Comment:
Minor: these nested switch/if statements look a bit odd to me. Could they maybe be combined?

Reply:
No easy way to do it. We could just use if elses?